tests/test_synthesize_sql_resources.py

```diff
@@ -0,0 +1,3427 @@
+                                                          new_json))
+
+    def test_003(self):
+        expected_sql = "SELECT * FROM test2, test1"
+        expected_json = {'from': ['test2', 'test1'], 'select': '*'}
+        new_sql = ""
+        new_json = ""
+        try:
+            new_sql = sql_synthesizer.generate_sql(expected_json)
+            new_json = parse(new_sql)
+            self.assertEqual(new_json,
+                             expected_json,
+                             self.get_error_message(expected_sql,
+                                                    new_sql,
+                                                    expected_json,
+                                                    new_json))
+        except Exception as e:
+            print(self.get_error_message(expected_sql, new_sql, expected_json, new_json))
+            print(e)
+            self.assertTrue(False, self.get_error_message(expected_sql,
+                                                          new_sql,
+                                                          expected_json,
+                                                          new_json))
+
+    def test_004(self):
+        expected_sql = "SELECT f1 FROM test1"
+        expected_json = {'from': 'test1', 'select': {'value': 'f1'}}
+        new_sql = ""
+        new_json = ""
+        try:
+            new_sql = sql_synthesizer.generate_sql(expected_json)
+            new_json = parse(new_sql)
+            self.assertEqual(new_json,
+                             expected_json,
+                             self.get_error_message(expected_sql,
+                                                    new_sql,
+                                                    expected_json,
+                                                    new_json))
+        except Exception as e:
+            print(self.get_error_message(expected_sql, new_sql, expected_json, new_json))
+            print(e)
+            self.assertTrue(False, self.get_error_message(expected_sql,
+                                                          new_sql,
+                                                          expected_json,
+                                                          new_json))
+
+    def test_005(self):
+        expected_sql = "SELECT f2 FROM test1"
+        expected_json = {'from': 'test1', 'select': {'value': 'f2'}}
+        new_sql = ""
+        new_json = ""
+        try:
+            new_sql = sql_synthesizer.generate_sql(expected_json)
+            new_json = parse(new_sql)
+            self.assertEqual(new_json,
+                             expected_json,
+                             self.get_error_message(expected_sql,
+                                                    new_sql,
+                                                    expected_json,
+                                                    new_json))
+        except Exception as e:
```

Comment:
Same goes for most of these other caught exceptions:  If you have three places where you call `self.get_error_message()` take you time to make it simpler.  At a minimum putting the logic into a function:

```python
def verify(expected_sql, expected_json):
       new_sql = ""
       new_json = ""
       try:
            new_sql = sql_synthesizer.generate_sql(expected_json)
            new_json = parse(new_sql)
            self.assertEqual(new_json,
                             expected_json,
                             self.get_error_message(expected_sql,
                                                    new_sql,
                                                    expected_json,
                                                    new_json))
        except Exception as e:
            print(self.get_error_message(expected_sql, new_sql, expected_json, new_json))
            print(e)
            self.assertTrue(False, self.get_error_message(expected_sql,
                                                          new_sql,
                                                          expected_json,
                                                          new_json))
```

but I stick with my suggestion of removing the `expect` and letting each test just fail.

Reply:
Hey, yes.  Thanks.  I had added extra output to help me debug the template, but agree it should really come out.